### ISSUE TITLE
RSS feed error in ownCloud log (modify header)

### ISSUE THREAD

**Issue by deMattin** [owncloud/activity#270]
I activated the activity rss link to be notified by rss.
To fetch all my rss feeds I use TinyTinyRSS.

Every time my TT-RSS calls owncloud rss feed, I get the following 6 errors in owncloud log:

>Cannot modify header information - headers already sent by (output started at /var/www/owncloud/lib/private/template/base.php:108) at /var/www/owncloud/lib/private/user/session.php#282
(line numbers in error log are: 282, 283, 284, 287, 288, 289)

If I use rss feed call `https://[domain]/owncloud/index.php/apps/activity/rss.php?token=kasdhfkhkasdkasdf123token` in a browser (IE11) **without** an active owncloud login cookie I get this errors in log, too.

If I call owncloud rss feed directly in browser **with** owncloud cookie set, I don't get this error message in log.

If I understand it right, the code in session.php linked to by the error message, deals with setting cookies, which shouldn't be done or tried by owncloud, if you call the rss feed with a token link!

RSS feed creation works and the feed is served - the issue is the log entry.

Edit:
Using: oC 8.0.2 (Suse repository) running on Debian 7

**Comment by nickvergessen** [comment]
@LukasReschke maybe we can have a look together at this next week

**Comment by deMattin** [comment]
Ok, should I remind you or do you set it on your todo for next week and review this, if your time allows it?

**Comment by deMattin** [comment]
Did you already have the chance to have a look at it?

**Comment by LukasReschke** [comment]
Cannot reproduce locally on stable8. Which apps do you have all enabled and could you provide us with your config? THX.

**Comment by nickvergessen** [comment]
Please also tell us the language/theme/... your instance uses

**Comment by deMattin** [comment]
Owncloud: 8.0.2-8 (suse repository)
Language: German
Theme: Tested both (with and without own theming)
Installed apps: files, activities, gallery+, rainloop mail, contacts, calendar, bookmarks, external apps

config:
`<?php
$CONFIG = array (
  'asset-pipeline.enabled' => false,
  'instanceid' => 'oc123blablablub',
  'passwordsalt' => '26blablablubblablablub1234123',
  'trusted_domains' => 
  array (
    0 => '192.168.192.210',
    1 => 'oc.mydomain.tld',
  ),
  'datadirectory' => '/media/sf_Owncloud-Data',
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => 'ocdb',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_myusers',
  'dbpassword' => 'hashofgeheim',
  'installed' => true,
  'default_language' => 'de',
  'forcessl' => true,
  'overwritehost' => 'oc.mydomain.tld',
  'overwriteprotocol' => 'https',
  'overwritewebroot' => '/owncloud',
  'overwrite.cli.url' => 'https://oc.mydomain.tld/owncloud',
  'trashbin_retention_obligation' => 180,
  'trashbin_auto_expire' => true,
  'loglevel' => 2,
  'logtimezone' => 'Europe/Berlin',
  'remember_login_cookie_lifetime' => 2592000,
  'enable_previews' => true,
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'share_folder' => '/__Shared__/',
  'theme' => '',
  'maintenance' => false,
  'secret' => 'aldskjhlaghlskfghnkjafo947thoaguo3478ghoau4hogaubl',
  'xframe_restriction' => false,
  'mail_smtpmode' => 'smtp',
  'mail_from_address' => 'xxxx',
  'mail_domain' => 'oc.mydomain.tld',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'smtp.xxxx.xxx',
  'mail_smtpport' => '587',
  'mail_smtpname' => 'xxxnamexxx',
  'mail_smtppassword' => 'xxxgeheimespasswortxxx',
  'mail_smtpsecure' => 'tls',
);`

I checked it again, because of the feedback, that it is not reproduceable.
I also deactivated my theming.
My oc instance is updated from a 7.x version of oc.
oC is running well with some little issues, I have read here in open or closed issues, that are not released..

I get the error with calling rss feed:
- with ie11 (without oc login cookie set)
- with debian install (iceweasel), where also oc webservice is installed
- with cron of my webservice to aggregate the news feeds (TTRSS)

I don't get the errors:
- with browser where oc login cookie is set

I don't fully understand the code but isn't there a check for login cookie if you call the rss feed?

The ress feed is generated and everything is working.
The error is just the error-code in log ... ;)

Thank for having a look at it.

**Comment by nickvergessen** [comment]
Can you try to view the page source when visiting it with the error?

**Comment by deMattin** [comment]
Are you referring the page source of rss feed?
There everything seems to be fine:
`https://sub.domain.tld/owncloud/index.php/apps/activity/rss.php?token=123456TokenVonOCVorgegeben7890`

Source of call with chrome with active login cookie: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
	<channel>
		<title>Feed der Aktivitäten</title>
		<language></language>
		<link>https://sub.domain.tld/owncloud/index.php/apps/activity/rss.php</link>
		<description>Persönlicher Feed der Aktivitäten für USER</description>
		<pubDate>Wed, 29 Apr 2015 14:32:05 +0000</pubDate>
		<lastBuildDate>Wed, 29 Apr 2015 14:32:05 +0000</lastBuildDate>
		<atom:link href="https://sub.domain.tld/owncloud/index.php/apps/activity/rss.php" rel="self" type="application/rss+xml" />
		<item>
			<guid isPermaLink="false">43922</guid>
			<title>Du hast Dokumente/blabla.xlsx geändert</title>
			<link>https://sub.domain.tld/owncloud/index.php/apps/files?dir=%2FDokumente</link>
			<pubDate>Wed, 29 Apr 2015 06:26:02 +0000</pubDate>
		</item>
		<item>
			<guid...
...
```
Source of call with ie11 without active login cookie (producing error in log) -> looks equal:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
	<channel>
		<title>Feed der Aktivitäten</title>
		<language></language>
		<link>https://sub.domain.tld/owncloud/index.php/apps/activity/rss.php</link>
		<description>Persönlicher Feed der Aktivitäten für USER</description>
		<pubDate>Wed, 29 Apr 2015 14:44:52 +0000</pubDate>
		<lastBuildDate>Wed, 29 Apr 2015 14:44:52 +0000</lastBuildDate>
		<atom:link href="https://sub.domain.tld/owncloud/index.php/apps/activity/rss.php" rel="self" type="application/rss+xml" />
		<item>
			<guid isPermaLink="false">43922</guid>
			<title>Du hast Dokumente/blabla.xlsx geändert</title>
			<link>https://sub.domain.tld/owncloud/index.php/apps/files?dir=%2FDokumente</link>
			<pubDate>Wed, 29 Apr 2015 06:26:02 +0000</pubDate>
		</item>
		<item>
			<guid ...
...
```
Activity page in oC itself gives no errors in Browser or log if I call it.
Maybe it's an issue only with my updated oC (I updated from oC 7.x).

If you are not ably to reproduce this in a new and actual instance of oC 8.0.x and also noone other can reproduce this, I think, it would be a good idea not to go in this issue deeper.

With new 8.1 release I will install oC completely new incl. new database, new user setup and on a fresh debian 8 and if I see this issue there too, I will reopen or renew this issue, if it is still a problem there.

I think, you already have reviewed the referenced lines in source code and if there should not be an issue there with these lines in rss feed call without a login cookie, my issue seems to be a individual case.

So from my side you may currently close this issue as an "individual install oC hickup issue" and spend you time on something more urgent ... ;)

**Comment by MorrisJobke** [comment]
@LukasReschke @nickvergessen What is the state of this?

**Comment by LukasReschke** [comment]
Still cannot reproduce. As discussed please file a new issue (or reopen) if this persists in 8.1 – strange issue… :speak_no_evil:

**Comment by deMattin** [comment]
I updated from 8.04 to 8.1 yesterday and don't see this errors any more.
Whatever problem has caused the errors, it seems to be fixed in 8.1.
